independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/logging/InitialConfigurator.java

```diff
@@ -14,25 +14,41 @@
 public final class InitialConfigurator implements EmbeddedConfigurator {
 
     public static final DelayedHandler DELAYED_HANDLER;
+    private static final Handler[] HANDLER_ARRAY;
 
     static {
-        //a hack around class loading
-        //this is always loaded in the root class loader with jboss-logmanager,
-        //however it may also be loaded in an isolated CL when running in dev
-        //or test mode. If it is in an isolated CL we load the handler from
-        //the class on the system class loader so they are equal
-        //TODO: should this class go in its own module and be excluded from isolated class loading?
-        DelayedHandler handler = new DelayedHandler();
+        boolean isQuarkusApplication = true;
+        DelayedHandler delayedHandler = new DelayedHandler();
         ClassLoader cl = InitialConfigurator.class.getClassLoader();
         try {
-            Class<?> root = Class.forName(InitialConfigurator.class.getName(), false, ClassLoader.getSystemClassLoader());
-            if (root.getClassLoader() != cl) {
-                handler = (DelayedHandler) root.getDeclaredField("DELAYED_HANDLER").get(null);
+            // Check and see if this is a Quarkus application. If it is, this class has already been loaded
+            Class.forName("io.quarkus.runner.ApplicationImpl", false, cl);
+
+            //a hack around class loading
+            //this is always loaded in the root class loader with jboss-logmanager,
+            //however it may also be loaded in an isolated CL when running in dev
+            //or test mode. If it is in an isolated CL we load the handler from
+            //the class on the system class loader so they are equal
+            //TODO: should this class go in its own module and be excluded from isolated class loading?
+            try {
+                Class<?> root = Class.forName(InitialConfigurator.class.getName(), false, ClassLoader.getSystemClassLoader());
+                if (root.getClassLoader() != cl) {
+                    delayedHandler = (DelayedHandler) root.getDeclaredField("DELAYED_HANDLER").get(null);
+                }
+            } catch (Exception e) {
+                //ignore, happens on native image build
             }
-        } catch (Exception e) {
-            //ignore, happens on native image build
+        } catch (ClassNotFoundException e) {
+            isQuarkusApplication = false;
```

Comment:
Shouldn't we have a better way to know that?

Reply:
I couldn't think of any given the fact that this class is loaded very very early...